fast64_internal/f3d/f3d_material.py

```diff
@@ -2288,6 +2326,51 @@ def draw(self, context):
 		ui_lower_mode(world.rdp_defaults, world, layout, True)
 		ui_other(world.rdp_defaults, world, layout, True)
 
+class CelLevelProperty(bpy.types.PropertyGroup):
+	inverse : bpy.props.BoolProperty(name = 'Inverse')
+	lighten : bpy.props.BoolProperty(name = 'Lighten')
+	fade : bpy.props.IntProperty(name = 'Fade', min = 0, max = 255, default = 200)
+	threshold : bpy.props.IntProperty(name = 'Threshold', min = 0, max = 255, default = 128)
+
+class CelShadingProperty(bpy.types.PropertyGroup):
+	solid : bpy.props.BoolProperty(name = "Color src: ENV (off = TEXEL0)")
+	levels : bpy.props.CollectionProperty(type = CelLevelProperty, name = "Cel Levels")
+
+def celGetMatlLevels(matlName):
+	for m in bpy.data.materials:
+		if m.name == matlName:
+			matl = m
+			break
+	else:
+		raise PluginError('Could not find material ' + matlName)
+	if m.mat_ver <= 3:
+		raise PluginError('Material version is <= 3')
+	return m.f3d_mat.cel_shading.levels
```

Comment:
shouldn't need to loop through everything here (also v3 will require conversion so no need to check for it)

```suggestion
def celGetMatlLevels(matlName):
	m = bpy.data.materials.get(matlName)
	if m is None:
		raise PluginError('Could not find material ' + matlName)
	return m.f3d_mat.cel_shading.levels
```

Reply:
Fixed